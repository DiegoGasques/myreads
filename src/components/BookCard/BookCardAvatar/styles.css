.book-card-avatar {
  min-width: 110px;
  min-height: 170px;

  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}
