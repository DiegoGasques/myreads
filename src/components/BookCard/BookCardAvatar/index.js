import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookCardAvatar({ src }) {
  return (
    <div
      className="book-card-avatar"
      style={{
        backgroundImage: `url(${src})`
      }}
    />
  );
}

BookCardAvatar.defaultProps = {
  src: "https://via.placeholder.com/100x170"
};

BookCardAvatar.propTypes = {
  src: PropTypes.string
};

export default BookCardAvatar;
