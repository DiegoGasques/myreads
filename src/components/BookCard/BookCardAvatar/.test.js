import React from "react";
import { shallow } from "enzyme";
import BookCardAvatar from "./index";
import toJSON from "enzyme-to-json";

it("Should match the component snapshot", () => {
  const wrapper = shallow(<BookCardAvatar />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
