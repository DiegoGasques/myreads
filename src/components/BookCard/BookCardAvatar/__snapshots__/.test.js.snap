// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="book-card-avatar"
  style={
    Object {
      "backgroundImage": "url(https://via.placeholder.com/100x170)",
    }
  }
/>
`;
